src/utils/index.ts

```diff
@@ -0,0 +1,73 @@
+// 1. result Input에 따라서 계산하는 함수 (Main)
+export const caculateResultInput = (resultInput: string) => {
+    const operatorIdx = findOperatorIdx(resultInput);
+    const operator = resultInput[operatorIdx];
+    const operatorPrevNumber = extractOperatorPrevNumber(resultInput, operatorIdx);
+    const operatorNextNumber = extractOperatorNextNumber(resultInput, operatorIdx);
+
+    return caculate(operator, operatorPrevNumber, operatorNextNumber);
+};
+// 2. result Input의 Operator를 위치를 구하는 함수
+const findOperatorIdx = (resultInput: string) => {
+    for (let i = 0; i < resultInput.length; i++) {
+        if (
+            resultInput[i] === '-' ||
+            resultInput[i] === 'X' ||
+            resultInput[i] === '+' ||
+            resultInput[i] === '/'
+        ) {
+            return i;
+        }
+    }
+    return -1;
+};
+// 3. Operator이전 숫자를 추출하기 위한 함수
+const extractOperatorPrevNumber = (
+    resultInput: string,
+    operatorIdx: number
+): number => {
+    return parseInt(resultInput.slice(0, operatorIdx));
+};
+
+// 4. Operator이후 숫자를 추출하기 위한 함수
+const extractOperatorNextNumber = (
+    resultInput: string,
+    operatorIdx: number
+): number => {
+    return parseInt(resultInput.slice(operatorIdx + 1, resultInput.length));
+};
+
+// 5. Operator이전 숫자와 이후 숫자를 가지고 연산하는 함수
+const caculate = (
+    operator: string,
+    one: number,
+    two: number
+): number | string => {
+    let calculateresultInput = 0;
+    switch (operator) {
+        case '+':
+            calculateresultInput = Math.floor(one + two);
+            break;
+        case '-':
+            calculateresultInput = Math.floor(one - two);
+            break;
+        case 'X':
+            calculateresultInput = Math.floor(one * two);
+            break;
+        case '/':
+            calculateresultInput = Math.floor(one / two);
+            break;
+    }
+    return !isValidResultInput(calculateresultInput) ? '오류' : calculateresultInput;
```

Comment:
값을 계산하는 로직하고 유한수인지 판단하는 후 반환을 하는 함수인데요.
어떻게 계산하는지에 대해서는 세부사항인 것 같아 별도 함수로 분리해보면 어떨까요?

```suggestion
    const value = getOperationResult({ operator, number1, number2 });
    return !isValidResultInput(value) ? '오류' : value;
```

아니면 LookUpTable을 이용해볼 수도 있을 것 같아요.
상수로 `Record<Operator, (number1: number, number2) => number>`을 만들어서 (key로 연산자, value로 각 키에 맞게 연산 후 반환)
```typescript
const value = operation[operator](number1, number2);
```
요렇게도 가능할 것 같습니다 ㅎ.ㅎ